src/main/java/org/ow2/proactive/procci/model/occi/platform/Component.java

```diff
@@ -33,14 +39,41 @@ public Component(Optional<String> url, Kind kind, Optional<String> title, List<M
         return attributeSet;
     }
 
-    public static class Builder extends Resource.Builder {
+    public static class Builder extends ResourceBuilder {
 
         protected Optional<Status> status;
 
         public Builder() {
             status = Optional.empty();
         }
 
+        public Builder(Model cloudAutomation) throws SyntaxException {
+            super(cloudAutomation);
+
+            Map<String, String> attributes = cloudAutomation.getVariables();
+            Optional<String> status = Optional.ofNullable(attributes.get(PlatformAttributes.STATUS_NAME));
+            if (status.isPresent()) {
+                this.status = Optional.of(Status.getStatusFromString(status.get()));
+            } else {
+                this.status = Optional.empty();
+            }
```

Comment:
Why not simply removing if ... else  and replacing 
"Optional<String> status = Optional.ofNullable(attributes.get(PlatformAttributes.STATUS_NAME));"
by
"Optional<String> status = Optional.ofNullable(attributes.getOrDefault(PlatformAttributes.STATUS_NAME), Status.getStatusFromString(status.get())); "